### DISCUSSION TITLE
404 while creating new PR review

### DISCUSSION THREAD

**Discussion by andre15silva** [community/community#24591] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I’m trying to use the PR review API for creating new reviews and new review comments.</p>
<p>The following request should work, but returns 404</p>
<pre><code class="lang-auto">❯ curl -X POST \
      -H "Accept: application/vnd.github.v3+json" -H "Authorization: token XXXXXXXX" \
      -H "Content-Type: application/json" \
      -d "{'body':'test', 'commit_id':'1b6897c66053c7fecf66853bdaea19a87b75c07b'}" \
      https://api.github.com/repos/andre15silva/failingProject/pulls/1/reviews
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/pulls#create-a-review-for-a-pull-request"
}
</code></pre>
<p>I have also tried using the official library <code>@octokit/request</code> as well as the unofficial library (which has this implemented) <code>hub4j/github-api</code> and both return 404.</p>
<p>The related PR is <a href="https://github.com/andre15silva/failingProject/pull/1" class="inline-onebox" rel="noopener nofollow ugc">NOMERGE: FL comments test PR by andre15silva · Pull Request #1 · andre15silva/failingProject · GitHub</a>, which does exists and should be available.</p>
<p>Am I missing something?</p>
<p>Best<br>
André</p>

**Comment by andre15silva** [comment, marked as answer]
<p>Hi,</p>
<p>So I finally found the issue. The token I was using didn’t have the permission <code>public_repo</code>, and so didn’t work.</p>
<p>This is really confusing tho, since the return code is <code>404</code> and no information about what permissions are required is given in the API documentation.</p>
<p>IMO, for this to be more user friendly, the required permissions should be explicit on the API documentation and the return code for the operation with a lack-privileged token should be 403.</p>

**Comment by jsoref** [comment]
<p>403 (and 410) aren’t used by a lot of systems because they leak the existence of things that someone might never have been entitled to know.</p>
<p>That said, some of the errors are indeed (dev) user hostile, and it’s possible that the linked page could note what permissions are generally required.</p>
